src/stores/widgets/StopGapWidget.ts

```diff
@@ -280,6 +281,10 @@ export class StopGapWidget extends EventEmitter {
         });
         this.messaging.on("capabilitiesNotified", () => this.emit("capabilitiesNotified"));
         this.messaging.on(`action:${WidgetApiFromWidgetAction.OpenModalWidget}`, this.onOpenModal);
+        this.messaging.on(`action:${ElementWidgetActions.JoinCall}`, () => {
+            // stop voice broadcast recording when joining a call
+            VoiceBroadcastRecordingsStore.instance().getCurrent()?.stop();
+        });
```

Comment:
It feels quite weird to be dealing with the underlying widget actions used by calls here, instead I would recommend putting this in the connect method of the Call class somewhere

Reply:
I will update the doc. Unfortunately `Call` is not involved for Jitsi widgets.

Reply:
Yes, but the JoinCall action isn't used by bare Jitsi widgets either (only by video rooms, which are Calls)

Reply:
At least my tests showed that Element Web Jitsi emits this event [here](https://github.com/vector-im/element-web/blob/develop/src/vector/jitsi/index.ts#L428).

Reply:
Oh, you're right, I failed to notice that this was looking for JoinCall actions sent by the widget to the client, rather than the other way around. In that case, this won't work with Element calls, since Element Call widgets don't respond with a JoinCall action, they only listen for incoming ones. Up to you whether it's important to make this work with Element calls

Reply:
Do they dispatch a `call_state` action? Then this would be covered over here https://github.com/matrix-org/matrix-react-sdk/pull/9425/files#diff-60dfd158c5a12a46d2c80ebd1ac9eacb189843e8f6b04e30d756e628dafa43d3R107

Reply:
They don't, no

Reply:
Good to know :thinking:

I would address Element Call calls in a separate PR. It is better to have this than nothing :D